backend/schema/jsonschema.go

```diff
@@ -88,21 +88,27 @@ func nodeToJSSchema(node Node, dataRefs map[DataRef]bool) *js.Schema {
 		st := js.Array
 		return &js.Schema{
 			Type:  &js.Type{SimpleTypes: &st},
-			Items: &js.Items{SchemaOrBool: &js.SchemaOrBool{TypeObject: nodeToJSSchema(node.Element, dataRefs)}},
+			Items: &js.Items{SchemaOrBool: &js.SchemaOrBool{TypeObject: nodeToJSSchema(node.Element, rootRef, dataRefs)}},
 		}
 
 	case *Map:
 		st := js.Object
 		// JSON schema generic map of key type to value type
 		return &js.Schema{
 			Type:                 &js.Type{SimpleTypes: &st},
-			AdditionalProperties: &js.SchemaOrBool{TypeObject: nodeToJSSchema(node.Value, dataRefs)},
-			PropertyNames:        &js.SchemaOrBool{TypeObject: nodeToJSSchema(node.Key, dataRefs)},
+			AdditionalProperties: &js.SchemaOrBool{TypeObject: nodeToJSSchema(node.Value, rootRef, dataRefs)},
+			PropertyNames:        &js.SchemaOrBool{TypeObject: nodeToJSSchema(node.Key, rootRef, dataRefs)},
 		}
 
 	case *DataRef:
-		ref := fmt.Sprintf("#/definitions/%s", node.String())
-		dataRefs[*node] = true
+		dataRef := *node
+		if dataRef.Module == "" {
+			// handle root data types
+			dataRef.Module = rootRef.Module
+		}
+
+		ref := fmt.Sprintf("#/definitions/%s", dataRef.String())
+		dataRefs[dataRef] = true
```

Comment:
I found that it was just generating a data ref for `Item` instead of `cart.Item` here so I added this. Maybe there's another approach, but this also seemed to resolve the error for me.